src/pipeline_factory.cpp

```diff
@@ -113,6 +113,7 @@ void PipelineFactory::revalidatePipelines(ModelManager& manager) {
 }
 const std::vector<std::string> PipelineFactory::getPipelinesNames() const {
     std::vector<std::string> names;
+    std::unique_lock lock(definitionsMtx);
```

Comment:
```
std::shared_lock lock(definitionsMtx);
```
This call can happen concurrently - we just need to be sure to not have new definitions created at the same time.